### PR TITLE
 client: remove unnecessary bufferptr[] for writev

### PR DESCRIPTION
**Pull request by batrick**
Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
Can you elaborate on the context a bit more in the commit message? Client is a big thing and I doubt you went through the whole code base looking for this pattern. ;)

File: src/client/Client.cc [hunk 1]
```diff
@@ -8781,19 +8781,19 @@ int Client::_write(Fh *f, int64_t offset, uint64_t size, const char *buf,
...
-  bufferptr *bparr = NULL;
+  std::vector<bufferptr> bpv;
   if (buf) {
       bufferptr bp;
       if (size > 0) bp = buffer::copy(buf, size);
       bl.push_back( bp );
   } else if (iov){
       //iov case 
-      bparr = new bufferptr[iovcnt];
+      bpv.resize(iovcnt);
```
Comment (batrick) [inline src/client/Client.cc]: I think you can but it's a C++14 thing according to:

http://en.cppreference.com/w/cpp/container/vector/vector

**Comment by batrick** [comment]
I improved the commit message @gregsfortytwo

**Comment by gregsfortytwo** [comment]
Good enough.

**Comment by batrick** [comment]
retest this please

**Comment by batrick** [comment]
retest this please

**Pull request merged**